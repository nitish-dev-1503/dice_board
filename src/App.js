import Tabs from './components/Tabs';

function App() {

  return (
    <div className="bg-neutral-200 text-primary_dark text-center">
      <header className="py-20">
        <p className='text-5xl font-bold'>Let's Play with Dice</p>
      </header>
      <Tabs />

      <p className="text-sm my-4 ">All rights reserved | Nitish Sharma</p>

    </div>
  );
}

export default App;
