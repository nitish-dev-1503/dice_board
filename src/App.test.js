import '@testing-library/jest-dom';
import { render, screen } from "@testing-library/react";
import renderer from 'react-test-renderer';
import App from "./App";


describe('App components snapshot', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<App />).toJSON();

        expect(tree).toMatchSnapshot();
    })
});

describe('App component', () => {
    test('renders logo', () => {
        render(<App />);

        const logo = document.querySelector('img');
        expect(logo.alt).toBe('logo');
    })

    test('renders title', () => {
        render(<App />);

        const title = screen.getByText('Let\'s Play with Dice', { exact: false });
        expect(title).toBeInTheDocument();
    })

    test('renders Tabs component', () => {
        const { getAllByRole } = render(<App />);

        const tabs = getAllByRole("listitem")
        expect(tabs.length).toBe(2);
    })

    test('renders footer', () => {
        render(<App />);

        const footer = screen.getByText('All rights reserved | Nitish Sharma');
        expect(footer).toBeInTheDocument();
    })
})