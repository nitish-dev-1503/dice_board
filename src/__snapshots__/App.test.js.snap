// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App components snapshot renders correctly 1`] = `
<div
  className="bg-neutral-100 text-primary_dark text-center"
>
  <header
    className="py-20"
  >
    <img
      alt="logo"
      className="h-30 my-4 m-auto"
      src="diceboard.svg"
    />
    <p
      className="text-xl font-bold "
    >
      ----- Let's Play with Dice -----
    </p>
  </header>
  <div>
    <ul
      className=" bg-primary_dark text-white font-medium flex w-1/2 m-auto items-center rounded-full"
    >
      <li
        className="tab  bg-primary"
        onClick={[Function]}
      >
        SevRoll
      </li>
      <li
        className="tab false"
        onClick={[Function]}
      >
        Reach50
      </li>
    </ul>
    <div>
      <div
        className="game"
      >
        <img
          alt="sevroll-logo"
          className="h-20 my-10"
          src="sevroll.svg"
        />
        <div
          className="text-left bg-teal-50 p-6 rounded-lg m-4"
        >
          <p
            className="font-medium"
          >
            How to play?
          </p>
          <p>
            Roll the dice to score points and score a multiple of 7 to win.
          </p>
        </div>
        <p
          className="my-2 font-medium"
        >
          Your Score
        </p>
        <p
          className=" text-6xl text-secondary font-bold "
        >
          0
        </p>
        <button
          className="btn-primary"
          disabled={false}
          onClick={[Function]}
        >
          Roll the dice
        </button>
        <p>
          roll the dice to play
        </p>
        <div
          className="h-20"
        />
        <button
          className="btn-primary"
          disabled={true}
          onClick={[Function]}
        >
          Replay
        </button>
      </div>
    </div>
  </div>
  <p
    className="text-sm py-4 "
  >
    All rights reserved | Nitish Sharma
  </p>
</div>
`;
